bsd/acl.pyx

```diff
@@ -259,53 +317,55 @@ cdef class ACLEntry(object):
         return {
             'tag': self.tag.name,
             'id': self.id,
-            'name': self.name,
             'type': self.type.name,
             'perms': {k.name: v for k, v in self.perms.items()},
             'flags': {k.name: v for k, v in self.flags.items()},
-            'text': self.text
         }
 
     def __setstate__(self, obj):
-        if 'text' in obj:
-            self.text = obj['text']
+        if 'tag' in obj:
+            self.tag = obj['tag']
 
-        if 'id' in obj:
+        if obj.get('id') is not None:
             self.id = obj['id']
 
-        if 'name' in obj:
-            self.name = obj['name']
-
         if 'type' in obj:
             self.type = obj['type']
 
         if 'perms' in obj:
-            for k, v in obj['perms']:
-                self.perms[k] = v
```

Comment:
I think the reason it was using a for loop was to copy the dict.
Just setting the attribute you're changing the behavior if you change the dict that was passed to bsd.acl as argument.

Reply:
tl;dr: dict is not a singleton